tracer/src/Datadog.Tracer.Native/cor_profiler.cpp

```diff
@@ -1299,6 +1299,39 @@ void CorProfiler::RegisterIastAspects(WCHAR** aspects, int aspectsLength)
     }
 }
 
+void CorProfiler::UpdateSettings(WCHAR* keys[], WCHAR* values[], int length)
+{
+    const WSTRING debugVarName = WStr("DD_TRACE_DEBUG");
+
+    for (int i = 0; i < length; i++)
+    {
+        if (WSTRING(keys[i]) == debugVarName)
```

Comment:
I'm not sure you need the `WSTRING(keys[i])`.
There is an overload of `==` operator which takes `WCHAR*` as parameter.

This `if (keys[i] == debugVarName)` will be fine and less allocations ;)